File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>> ans;
    
        for(int i=0;i<strs.size();i++){
            string s= strs[i];
            sort(strs[i].begin(),strs[i].end());
            ans[strs[i]].push_back(s);
        }

        vector<vector<string>> a;
        for(auto it:ans){
            
            a.push_back(it.second);
        }
        return a;

    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        mp = {}
        for i in range(0,len(strs)):
            temp = ''.join(sorted(strs[i]))
            if temp not in mp:
                mp[temp]=[]
            mp[temp].append(i)
        ans = []

        for i in mp:
            temp=[]
            for j in mp[i]:
                temp.append(strs[j])
            ans.append(temp)
        return ans


        